support/reftest/reftest

```diff
@@ -0,0 +1,249 @@
+    )
+
+
+def parse_cdn_certs(parser):
+    parser.add_argument(
+        "--cert",
+        default="~/certs/rcm-debug/rcm-debug.crt",
+        type=str,
+        help="Client certificate file and password",
+    )
+    parser.add_argument(
+        "--key",
+        default="~/certs/rcm-debug/rcm-debug.key",
+        type=str,
+        help="Private key file name",
+    )
+    parser.add_argument(
+        "--cacert",
+        default="/etc/rhsm/ca/redhat-uep.pem",
+        type=str,
+        help="CA certificate to verify peer against",
+    )
+
+
+def parse_args():
+    root_parser = argparse.ArgumentParser(description=__doc__)
+    subparsers = root_parser.add_subparsers(
+        dest="command", help="reference test data operations"
+    )
+    # parser for prepare operation
+    parser_prepare = subparsers.add_parser(
+        "prepare",
+        help="""
+        prepare reference test data in dynamodb and s3
+        for integration test,
+         e.g. $./reftest prepare --bucket exodus-bucket --table exodus-table
+        """,
+    )
+    parser_prepare.add_argument(
+        "--release-date",
+        default=None,
+        help="Date on which the content will be made available.",
+    )
+    parse_aws_session(parser_prepare)
+    parse_cdn_certs(parser_prepare)
+    parser_prepare.add_argument(
+        "--bucket",
+        required=True,
+        help="The AWS S3 bucket used to store test data",
+    )
+    parser_prepare.add_argument(
+        "--table",
+        required=True,
+        help="The AWS dynamoDB used to store test data",
+    )
+
+    return root_parser.parse_args()
+
+
+class RefTestConfig:
+    def __init__(self, prod_cdn_url, test_data):
+        self.prod_cdn_url = prod_cdn_url
+        self.test_data = test_data
+
+
+def load_config():
+    with open("data.yml") as f:
+        config = yaml.load(f, yaml.SafeLoader)
+    return RefTestConfig(config["prod-cdn-url"], config["test_data"],)
+
+
+def download_to_local(url, key_path, cert_path, cacert_path):
+    with requests.get(
+        url,
+        cert=(cert_path, key_path),
+        verify=cacert_path,
+        stream=True,
+        timeout=(30, 30),
+    ) as r:
+        r.raise_for_status()
+        total_size = int(r.headers.get("content-length", 0))
+        t = tqdm(desc=url, total=total_size, unit="iB", unit_scale=True)
+
+        # the file is deleted as soon as it is closed.
+        temp_file = tempfile.NamedTemporaryFile(delete=True)
+        for chunk in r.iter_content(chunk_size=8192):
+            if chunk:  # filter out keep-alive new chunks
+                t.update(len(chunk))
+                temp_file.write(chunk)
+                temp_file.flush()
+        t.close()
+        return temp_file
+
+
+# verify cdn data checksum mathch test data checksum
+def verify_sha256_checksum(temp_file_path, test_data_checksum):
+    if os.path.isfile(temp_file_path):
+        sha256 = hashlib.sha256()
+        with open(temp_file_path, "rb") as binary:
```

Comment:
It's not really a big deal, but this is wasteful because you already read the entire file (while downloading it) and now you're reading the whole thing again.

Nathan already pointed out above in the comment "Looks like a good opportunity [...]" exactly where we could calculate checksum.  Doing it there would avoid having to re-read the file we've just written unnecessarily.

That being said, I doubt it has any notable impact on performance.